// Operations that can be performed on entities.

syntax = "proto3";

package inhumate.rti.proto;
option csharp_namespace = "Inhumate.RTI.Proto";

import "google/protobuf/empty.proto";

message EntityOperation {
    oneof which {
        google.protobuf.Empty request_update = 1;
        ApplicationClient request_persistent_ownership = 2;
        ApplicationClient claim_persistent_ownership = 3;
        EntityClient transfer_ownership = 4;
        EntityClient assume_ownership = 5;
        EntityClient release_ownership = 6;
    }

    message EntityClient {
        string entity_id = 1;
        string client_id = 2;
    }

    message ApplicationClient {
        string application = 1;
        string client_id = 2;
    }

}
